node/node.go

```diff
@@ -1379,23 +1388,44 @@ func createAndStartPrivValidatorSocketClient(
 	return pvscWithRetries, nil
 }
 
-// splitAndTrimEmpty slices s into all subslices separated by sep and returns a
-// slice of the string s with all leading and trailing Unicode code points
-// contained in cutset removed. If sep is empty, SplitAndTrim splits after each
-// UTF-8 sequence. First part is equivalent to strings.SplitN with a count of
-// -1.  also filter out empty strings, only return non-empty strings.
-func splitAndTrimEmpty(s, sep, cutset string) []string {
-	if s == "" {
-		return []string{}
-	}
-
-	spl := strings.Split(s, sep)
-	nonEmptyStrings := make([]string, 0, len(spl))
-	for i := 0; i < len(spl); i++ {
-		element := strings.Trim(spl[i], cutset)
-		if element != "" {
-			nonEmptyStrings = append(nonEmptyStrings, element)
-		}
+func createAndStartPrivValidatorGRPCClient(
+	config *cfg.Config,
+	chainID string,
+	logger log.Logger,
+) (types.PrivValidator, error) {
+	listenAddr := config.PrivValidatorListenAddr
```

Comment:
This is a bit odd to me. A listen address is typically when _we_ listen on a port, not some address that someone _else_ is listening on and that we are going to dial. I suppose this follows from the inversion of the server/client roles in the old model and the gRPC model.

I know this is temporary until we remove the old protocol, so maybe we can clean this up later, but this should really be called something else.

Reply:
Will remove the old protocol in a follow up PR, this PR was centered around providing a well reviewed PR to backport.